src/lib.rs

```diff
@@ -687,6 +703,27 @@ impl<'a> ParserState<'a> {
         }
     }
 
+    /// An MD5 mangled name is `??@` followed by 32 characters and a terminating `@`.
+    ///
+    /// See https://github.com/llvm/llvm-project/blob/818cf30b83305fa4a2f75821349210b0f7aff4a4/llvm/lib/Demangle/MicrosoftDemangle.cpp#L754
+    fn read_md5_name(&mut self) -> Result<Name<'a>> {
+        let start_offset = self.offset;
+
+        if start_offset < 3 || !self.input[start_offset - 3..].starts_with("??@") {
+            return Err(self.fail("expected MD5 mangled name"));
+        }
+
+        while !self.consume(b"@") {
+            self.advance(1);
+        }
+        if self.offset - start_offset != 33 {
+            return Err(self.fail("expected MD5 mangled name of length 32"));
+        }
+        Ok(Name::Md5(
+            &self.input.as_bytes()[start_offset - 3..self.offset],
```

Comment:
Does this slice include the `??@` at the beginning? If so, why? Oh, so that it's output unchanged? Hmm, I think it would be slightly nicer to have just the hex characters in this slice, and then put the `??@...@` around them in the output functions.